tensorboard/components/tf_runs_selector/tf-wbr-string.html

```diff
@@ -0,0 +1,75 @@
+<!--
+@license
+Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<link rel="import" href="../tf-imports/polymer.html" />
+
+<!--
+  tf-wbr-string safely renders a string, with <wbr> elements inserted
+  around select delimiters.
+-->
+<dom-module id="tf-wbr-string">
+  <template>
+    <!--
+      This ugly formatting is required to prevent spaces from slipping
+      into the HTML.
+    -->
+    <template is="dom-repeat" items="[[_parts]]" as="part"
+      >[[part.contents]]<template is="dom-if" if="[[!part.last]]"
+        ><wbr /></template
+    ></template>
+  </template>
+  <script>
+    Polymer({
+      is: 'tf-wbr-string',
+      properties: {
+        /** The value to render with word breaks. */
+        value: String,
+        /**
+         * Insert a word-break after each match of this regex.
+         *
+         * This `RegExp` object will not be mutated.
+         */
+        delimiterPattern: {
+          type: Object /* RegExp */,
+          value: /([\/=\-_,])/,
+        },
+        _parts: {
+          type: Array /* {contents: string, last: boolean}[] */,
+          computed: '_computeParts(value, delimiterPattern)',
+        },
+      },
+      _computeParts(value, delimiterPattern) {
+        const result = [];
+        if (value == null) {
+          value = '';
+        }
+        while (true) {
+          const idx = value.search(delimiterPattern);
+          if (idx === -1) {
+            result.push({contents: value, last: true});
+            break;
+          } else {
+            const part = value.slice(0, idx + 1);
```

Comment:
Isn't `idx + 1` only correct if the delimiter match is exactly 1 character?  If that's all we want to support it should be documented as such, and then in that case maybe we should construct the regex ourselves to control for this.

Reply:
Yes; good point. I’ll make the pattern private.